### PR TITLE
feat: validate requiresSyncData keys

### LINKED ISSUE #264: Requiressync annotation allows GVKs with multiple versions
Is the schema for allowing multiple versions listed in the same GVK ([example](https://github.com/open-policy-agent/gatekeeper-library/blob/master/library/general/uniqueingresshost/template.yaml#L18)) set in stone? I think it makes unmarshaling the json into the [gvk type](https://pkg.go.dev/sigs.k8s.io/kustomize/pkg/gvk) a little less convenient. I'm happy to open a PR to change the templates in the library if this change is agreed upon. But I'm also happy to implement custom unmarshaling logic for my use case if the existing schema is a necessity.

### PR DESCRIPTION
**Pull request by apeabody**
Signed-off-by: Andrew Peabody <andrewpeabody@google.com>

**What this PR does / why we need it**:
Prevent regression of https://github.com/open-policy-agent/gatekeeper-library/issues/264#issuecomment-1349020755

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #264 

**Special notes for your reviewer**:
https://github.com/open-policy-agent/gatekeeper-library/pull/269 needs to be merged first (CI)

### CODE REVIEW

File: scripts/require-sync/main.go [hunk 1]
```diff
@@ -191,3 +199,34 @@ func opaClient(referential bool) (*opa.Client, error) {
...
+	}
+
+	// Unmarsall JSON
+	var contents []interface{}
+	if err := json.Unmarshal([]byte(annotation), &contents); err != nil {
+		return false, fmt.Errorf("Error validating JSON content")
+	}
+
+	// Validate keys
+	for _, value := range contents {
```
Comment (maxsmythe) [inline scripts/require-sync/main.go:221]: nit: we should change the variable names to make intent clear and avoid confusion around shadowing.

Maybe: outer-loop-var => `requirement`, inner-loop-var => `equivalents`?

**Review by maxsmythe — APPROVED** [review]
LGTM

**Comment by sozercan** [comment]
@apeabody looks like ci is failing

**Comment by apeabody** [comment]
> @apeabody looks like ci is failing. is this due to #269 prereq?

Hi @sozercan - Yes, a failure in that check is expected untill #269 is merged.

File: scripts/require-sync/main.go (hunk 1 above)
Comment (apeabody) [reply scripts/require-sync/main.go]: Sure, those sound good to me.

File: scripts/require-sync/main.go [hunk 2]
```diff
@@ -191,3 +199,34 @@ func opaClient(referential bool) (*opa.Client, error) {
 
 	return client, nil
 }
+
+func validateRequiresSyncDataContent(annotation string) (bool, error) {
```
Comment (sozercan) [inline scripts/require-sync/main.go:203]: can we add a test please

File: scripts/require-sync/main.go [hunk 3]
```diff
@@ -191,3 +199,34 @@ func opaClient(referential bool) (*opa.Client, error) {
...
+
+	// Remove outer quotes
+	annotation = annotation[1 : len(annotation)-1]
+
+	// Validate JSON
+	if ok := json.Valid([]byte(annotation)); !ok {
+		return false, fmt.Errorf("Error validating JSON format")
+	}
+
+	// Unmarsall JSON
```
Comment (sozercan) [inline scripts/require-sync/main.go:214]:
```suggestion
	// Unmarshal JSON
```

Comment (apeabody) [reply scripts/require-sync/main.go]: done

File: scripts/require-sync/main.go (hunk 2 above)
Comment (apeabody) [reply scripts/require-sync/main.go:203]: Sure - I added a simple test in `main_test.go`

**Comment by apeabody** [comment]
Hi @sozercan - Hope you had a nice holiday, anything else needed on this change?  Thanks!

**Review by sozercan — APPROVED** [review]
Sorry for the delay and thank you for adding this, LGTM

**Pull request merged**